useIsShake.js

```diff
@@ -1,59 +1,44 @@
 import React, { useState, useEffect } from 'react';
 import { Accelerometer } from 'expo-sensors';
-import { QUOTES_IN_DURATION, QUOTES_IN_DELAY, QUOTES_OUT_DURATION, SVG_LOAD_DURATION, SVG_OUT_DURATION, SVG_IN_DURATION, SVG_IN_DELAY, TRANSITION_TIME_2 } from './constants';
 
 export function useIsShake() {
   const [{ x }, setData] = useState({
     x: 0,
   });
   const [subscription, setSubscription] = useState(null);
-  const [toggle, setToggle] = useState(false);
+  const [isShakeTriggered, setIsShakeTriggered] = useState(false);
+  const [isToggleReady, setIsToggleReady] = useState(false);
 
   let polls = [null, null];
   let diffs = [];
-  let shakeCount = 0;
-  let isToggleReady;
-  let hasToggledBefore;
 
   function isShake({ x: newX }) {
-    if (isToggleReady === true) {
-      if (diffs.length === 2) {
-        shakeCount++;
-        setToggle(true);
-        isToggleReady = false;
+    // console.log(newX);
 
-        if (hasToggledBefore === undefined) hasToggledBefore = false;
-        else if (hasToggledBefore === false) hasToggledBefore = true;
+    if (diffs.length === 2) {
+      setIsShakeTriggered(true);
+      setIsToggleReady(false);
 
-        let debounceTime;
-        if (hasToggledBefore === false) {
-          debounceTime = Math.max(SVG_OUT_DURATION, (QUOTES_IN_DURATION + QUOTES_IN_DELAY));
-        } else {
-          debounceTime = TRANSITION_TIME_2;
-        }
+      console.log("isShakeTriggered", { diffs });
 
-        const toReady = setTimeout(() => {
-          setToggle(false);
-          isToggleReady = true;
-          clearTimeout(toReady);
-        }, debounceTime);
```

Comment:
Redesigned implementation of debounce functionality and no longer using setTimeout to end debounce time.